docs/_docs/04_reference/02-cli-options.md

```diff
@@ -661,6 +666,117 @@ By default, hidden directories (those starting with `.`) are excluded from the s
 terragrunt find --hidden
 ```
 
+#### list
+
+**[NOTE] The `list` command is experimental, usage requires the [`--experiment cli-redesign` flag](/docs/reference/experiments/#cli-redesign).**
+
+List Terragrunt configurations in your codebase.
+
+```bash
+terragrunt list [options]
+```
+
+The `list` command helps you discover and display Terragrunt configurations in your codebase. It provides various output formats and options to help you understand the structure and dependencies of your Terragrunt configurations.
+
+The `list` command is very similar to the [`find` command](/docs/reference/cli-options/#find). It uses the same backend logic for configuration discovery. It also supports very similar flags and configuration options.
+
+For example, you can use the `--format` and `--sort` flags to adjust the format of output, and the sorting of results, respectively. The `list` command also supports the `--dependencies` flag to include dependency information in output when available, the `--external` flag to include external dependencies, and the `--hidden` flag to include hidden configurations.
+
+It also uses global flags, such as `--working-dir` and `--no-color` to control behavior, and will avoid emitting color output when stdout is redirected. To save space here, common behavior between the `find` and `list` commands may be omitted from the `list` documentation. This documentation will focus on the differentiating features of the `list` command.
+
+Generally, the `list` command is optimized for displaying configurations in a format that is easy for a human to read and understand, so there are more configuration options, and the output is more verbose. Output from the `list` command can vary significantly depending on the flags used, and the output structure is more variable than that of the `find` command.
```

Comment:
You hint at it, but we should be explicit when we recommend using `list` vs. `find`.

Reply:
Made it more explicit.